libraries/networking/src/NLPacket.h

```diff
@@ -32,27 +38,29 @@ class NLPacket : public Packet {
     virtual qint64 localHeaderSize() const;  // Current level's header size
 
     const QUuid& getSourceID() const { return _sourceID; }
-    const QByteArray& getVerificationHash() const { return _verificationHash; }
     
     void writeSourceID(const QUuid& sourceID);
-    void writeVerificationHash(const QByteArray& verificationHash);
-
-    QByteArray payloadHashWithConnectionUUID(const QUuid& connectionUUID) const;
+    void writeVerificationHashGivenSecret(const QUuid& connectionSecret);
 
 protected:
     
-    void adjustPayloadStartAndCapacity();
+    void adjustPayloadStartAndCapacity(bool shouldDecreasePayloadSize = false);
     
     NLPacket(PacketType::Value type);
     NLPacket(PacketType::Value type, qint64 size);
+    
     NLPacket(std::unique_ptr<char> data, qint64 size, const HifiSockAddr& senderSockAddr);
+    NLPacket(std::unique_ptr<Packet> packet);
+    
     NLPacket(const NLPacket& other);
+    NLPacket& operator=(const NLPacket& other);
+    
+    NLPacket(NLPacket&& other);
```

Comment:
Guess I broke it out because I thought the move constructor is special but that probably makes sense